public/app/features/scopes/selector/types.ts

```diff
@@ -1,35 +1,16 @@
-import { Scope, ScopeNodeSpec } from '@grafana/data';
+import { Scope, ScopeNode } from '@grafana/data';
 
-export enum NodeReason {
-  Persisted,
-  Result,
-}
-
-export interface Node extends ScopeNodeSpec {
-  name: string;
-  reason: NodeReason;
-  expandable: boolean;
-  selectable: boolean;
-  expanded: boolean;
-  query: string;
-  nodes: NodesMap;
-}
-
-export type NodesMap = Record<string, Node>;
+export type NodesMap = Record<string, ScopeNode>;
```

Comment:
```suggestion
export type NodesMap = Record<string, ScopeNode | undefined>;
```
To make sure that we always check for its existence. Maybe we should do the same for ScopesMap? I'm also wondering if we should use a `Map` instead?

Reply:
This creates problem with `Object.values(nodesMap)` which that thinks those can be undefined and also allows you to do `nodesMap['somehting'] = undefined` which isn't ideal. So Map would probably be better here but considering it changes the way values are accessed maybe I would leave it for later PR